Library/Homebrew/os/linux/elf.rb

```diff
@@ -89,13 +96,16 @@ def with_interpreter?
   def dynamic_elf?
     return @dynamic_elf if defined? @dynamic_elf
 
-    @dynamic_elf = if which "readelf"
-      Utils.popen_read("readelf", "-l", to_path).include?(" DYNAMIC ")
-    elsif which "file"
-      !Utils.popen_read("file", "-L", "-b", to_path)[/dynamic|shared/].nil?
-    else
-      raise "Please install either readelf (from binutils) or file."
-    end
+    @dynamic_elf =
+      if HOMEBREW_PATCHELF_RB
+        patcher.instance_variable_get(:@elf).segment_by_type(:DYNAMIC).present?
```

Comment:
```suggestion
        patchelf_patcher.instance_variable_get(:@elf).segment_by_type(:DYNAMIC).present?
```

Reply:
Also it's pretty weird to get an instance variable like this. Would be nice to submit a PR upstream to make this an accessible by a method.

Reply:
ohkay, i'll patch and request a merge.

Reply:
a pull request has been submitted to upstream, should be merged soon.